libcudacxx/include/cuda/std/detail/libcxx/include/__type_traits/promote.h

```diff
@@ -27,12 +27,32 @@
 #include "../__utility/declval.h"
 #include "../cstddef"
 
+#if defined(_LIBCUDACXX_CUDACC)
```

Comment:
I am still curious about that constraint here, as it is different from all other constraints we have in the code.

Reply:
Yeah, I don't think this is needed anymore.